yt_dlp/extractor/videocampus_sachsen.py

```diff
@@ -162,7 +181,86 @@ def _real_extract(self, url):
+    _VALID_URL = r'''(?x)(?P<host>https?://(?:%s))/(?:
+        album/view/aid/(?P<album_id>[0-9]+)|(?P<c_mode>category|channel)/(?P<c_name>[\w-]+)/(?P<c_id>[0-9]+)
+        )''' % ('|'.join(map(re.escape, VideocampusSachsenIE._INSTANCES)))
+
+    _TESTS = [{
+        'url': 'https://vimp.oth-regensburg.de/channel/Designtheorie-1-SoSe-2020/3',
+        'info_dict': {
+            'id': 'channel-3',
+            'title': 'Designtheorie 1 SoSe 2020 :: Channels :: ViMP OTH Regensburg',
+        },
+        'playlist_mincount': 9,
+    }, {
+        'url': 'https://www.fh-bielefeld.de/medienportal/album/view/aid/208',
+        'info_dict': {
+            'id': 'album-208',
+            'title': 'KG Praktikum ABT/MEC :: Playlists :: FH-Medienportal',
+        },
+        'playlist_mincount': 4,
+    }, {
+        'url': 'https://videocampus.sachsen.de/category/online-tutorials-onyx/91',
+        'info_dict': {
+            'id': 'category-91',
+            'title': 'Online-Seminare ONYX - BPS - Bildungseinrichtungen - VCS',
+        },
+        'playlist_mincount': 7,
+    }]
+
+    def _real_extract(self, url):
+        host, album_id, c_mode, c_name, c_id = self._match_valid_url(url).group(
+            'host', 'album_id', 'c_mode', 'c_name', 'c_id')
+
+        webpage = self._download_webpage(url, album_id or c_id, fatal=False) or ''
+        title = (self._html_search_meta('title', webpage, fatal=False)
+                 or self._html_extract_title(webpage))
+
+        if album_id:
+            mode = 'album'
+            mode_id = '4'
+            content_id = album_id
+            url_part = f'aid/{album_id}'
+        else:
+            url_parts = ('category', 'category_id') if c_mode == 'category' else ('title', 'channel')
+
+            mode = c_mode
+            mode_id = '1' if c_mode == 'category' else '3'
+            content_id = c_id
+            url_part = f'{url_parts[0]}/{c_name}/{url_parts[1]}/{c_id}'
+
+        data = {
+            'vars[mode]': mode,
+            f'vars[{mode}]': content_id,
+            'vars[context]': mode_id,
+            'vars[context_id]': content_id,
+            'vars[layout]': 'thumb',
+            'vars[per_page][thumb]': '10',
+        }
+
+        urls, i = [], 1
+        while True:
+            current_page = self._download_webpage(
+                f'{host}/media/ajax/component/boxList/{url_part}?page={i}&page_only=1',
+                content_id, data=urlencode_postdata(data))
+            current_urls = re.findall(r'"(.+?/video/.+?)"', current_page)
+
+            # Request returns last n video URLs if existing pages have been "used up"
+            if urls[-1:] != current_urls[-1:]:
+                urls += current_urls
+                i += 1
+            else:
+                break
+
+        return self.playlist_result(
+            (self.url_result(host + url, VideocampusSachsenIE) for url in urls),
+            playlist_title=title, id=f'{mode}-{content_id}')
```

Comment:
ViMP has three "playlist modes": albums, categories, channels. Each of them has it's own set of identifiers (both `album-1` and `channel-1` can exist).

Is it OK to self-buid the `id` here? Should it be constructed differently?

Reply:
playlist ids don't matter much, as long as they are sensible enough